MediaBrowser.LocalMetadata/Savers/BaseXmlSaver.cs

```diff
@@ -94,6 +94,14 @@ public async Task SaveAsync(BaseItem item, CancellationToken cancellationToken)
         {
             var path = GetSavePath(item);
             var directory = Path.GetDirectoryName(path) ?? throw new InvalidDataException($"Provided path ({path}) is not valid.");
+
+            if (directory.Length > 260 || item.Name.Length > 255)
+            {
+                _ = OperatingSystem.IsWindows() ? _ = directory.Insert(0, @"\\?\") : directory = Path.Combine(
```

Comment:
I dont know if its handled differently on MacOS did you check that too?

Reply:
also please expand that code, its very hard to read

Reply:
Updated just now, see comments in most recent commit. Any metadata which doesn't inherit from Folder gets its save file name from an existing file, preventing the possibility of it exceeding the maximum path length.